woocommerce-gateway-stripe.php

```diff
@@ -390,3 +390,34 @@ function woocommerce_gateway_stripe_init() {
 
 	woocommerce_gateway_stripe();
 }
+
+// Hook in Blocks integration. This action is called in a callback on plugins loaded, so current Stripe plugin class
+// implementation is too late.
+add_action( 'woocommerce_blocks_loaded', 'woocommerce_blocks_support' );
+
+function woocommerce_blocks_support() {
+	if ( class_exists( 'Automattic\WooCommerce\Blocks\Payments\Integrations\AbstractPaymentMethodType' ) ) {
+		require_once dirname( __FILE__ ) . '/includes/class-wc-stripe-blocks-support.php';
+		// priority is important here because this ensures this integration is
+		// registered before the WooCommerce Blocks built-in Stripe registration.
+		// Blocks code has a check in place to only register if 'stripe' is not
+		// already registered.
+		add_action(
+			'woocommerce_blocks_payment_method_type_registration',
+			function( Automattic\WooCommerce\Blocks\Payments\PaymentMethodRegistry $payment_method_registry ) {
+				$container = Automattic\WooCommerce\Blocks\Package::container();
+				// registers as shared instance.
+				$container->register(
+					WC_Stripe_Blocks_Support::class,
+					function() {
+						return new WC_Stripe_Blocks_Support;
+					}
+				);
+				$payment_method_registry->register(
+					$container->get( WC_Stripe_Blocks_Support::class )
+				 );
+			},
+			5
```

Comment:
Thanks to [this code](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/06638927bfa3ff6dd4ed92b5742411be16d42ec7/src/Payments/Api.php#L108-L112), by registering this integration earlier than the Blocks integration (which uses the same hook at priority 10), this basically ensures that the Stripe extension is what get's used for the integration rather than what is embedded in WooCommerce Blocks.